### PR TITLE
Adding SwiftLint and resolve warnings/errors

### LINKED ISSUE #218: Configure linter
And integrate in CI process. Linter errors should fail the build and not allow to merge PRs.

### PR DESCRIPTION
**Pull request by astewart43-gtri**
closes #218

### CODE REVIEW

**Comment by ivnsch** [comment]
Small request in advance: Can we disable the identifiers min chars rule? This is IMO silly as it forbids to use e.g. `id`. Variables and method names with only one character should also be possible (we use this e.g. for the logs).

**Comment by astewart43-gtri** [comment]
Yes, the latest push fixes that.
min length is set to 1 max is set to 60.

**Comment by ivnsch** [comment]
Nice! Sure that length 1 isn't causing a warning though?

**Comment by astewart43-gtri** [comment]
@i-schuetz This seems to be a ">=1" check. So no, it is working as desired.

File: .swiftlint.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
+disabled_rules:
+- trailing_whitespace
```
Comment (ivnsch) [inline .swiftlint.yml]: Why is this disabled?

File: .swiftlint.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+- line_length
```
Comment (ivnsch) [inline .swiftlint.yml]: Why disabled? It's used below?

File: .swiftlint.yml (hunk 1 above)
Comment (astewart43-gtri) [reply .swiftlint.yml]: My mistake, will correct.

File: .swiftlint.yml (hunk 2 above)
Comment (astewart43-gtri) [reply .swiftlint.yml]: My mistake. I was testing and forgot to revert. Will correct.

File: .swiftlint.yml [hunk 3]
```diff
@@ -0,0 +1,40 @@
+disabled_rules:
+- nesting
+- unused_closure_parameter
```
Comment (ivnsch) [inline .swiftlint.yml]: Why was this one added?

Comment (astewart43-gtri) [reply .swiftlint.yml]: nesting per our slack conversation (//Nesting Violation: Types should be nested at most 1 level deep (nesting))

unused_closure_parameter was a mistake. It needed to be aded while the auto corrector was running. it has been removed.

**Comment by astewart43-gtri** [comment]
I made a best effort to go through the code and clean up inconsistent formatting. I likely missed a lot, but I think this is a good starting point.

**Comment by ivnsch** [comment]
Meaning we have a lot of new warnings now? Can you tackle those in a new PR?

**Comment by astewart43-gtri** [comment]
most of the existing warnings are related to TODO comments, force_cast, and force_try.
yes, everything can be addressed in a separate PR, assuming the formatting is what you want it to be.
like the function names that were too long, separating them into multiple lines in a way that makes sense.

**Review by ivnsch — APPROVED** [review]

**Comment by duskoo** [comment]
@astewart43-gtri Current develop (c36a5bd) is not buildable due to too many swiftlint errors. Any suggestions?

**Comment by astewart43-gtri** [comment]
it seems to build fine for me. 
However, if you need to ignore an error, you can add it under "disabled_rules:" heading in .swiftlint.yml

**Pull request merged**